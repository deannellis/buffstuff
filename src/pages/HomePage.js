import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Link} from "react-router-dom";
import moment from 'moment';

import { startWorkout } from '../actions/inProgressWorkout';
import { setTargetMacros, updateMacro, setCurrentDate, resetCurrent } from '../actions/macros';
import Dashboard from '../components/Dashboard';
import Button from '../components/Button';
import SideNav from '../components/SideNav';

class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            today: moment()
        }
    }

    onStartWorkout = () => {
        this.props.dispatch(startWorkout(this.props.liftVariant));
        this.props.history.push('/workout');
    }

    onSetMacros = macros => { this.props.dispatch(setTargetMacros(macros)) }

    onUpdateMacro = update => {
        this.props.dispatch(updateMacro(update))
    }

    componentDidMount() {
        const { dateObject } = this.props.macros;
        const storedDate = dateObject._d
        const { today } = this.state;
        if(dateObject === null || dateObject === 0) {
            console.log('SETTING DATE DUE TO NULL OR 0 VALUE');
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }else if(!(moment().isSame(storedDate, 'day'))) {
            console.log('RESETING CURRENT MACROS BASED ON OUTDATED DATE');
            this.props.dispatch(resetCurrent());
            this.props.dispatch(setCurrentDate({ currentDate: today }));
        }
    }

    render() { 
        if(Object.entries(this.props.masterWeights).length === 0 && this.props.masterWeights.constructor === Object) {
            return (
                <>
                    <h2>Welcome to flext!</h2>
                    <p>Let's begin by entering your starting weights</p>
                    <Link to="/onboarding">
                        <Button variant="primary">
                            Enter Weights
                        </Button>
                    </Link>
                </>
            );
        }
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <Dashboard
                    liftVariant={this.props.liftVariant}
                    masterWeights={this.props.masterWeights} 
                    workouts={this.props.workouts}
                    onStartWorkout={this.onStartWorkout}
                    macros={this.props.macros}
                    onSetMacros={this.onSetMacros}
                    onUpdateMacro={this.onUpdateMacro}
                />
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
        liftVariant: state.liftVariant,
        workouts: state.workouts,
        macros: state.macros,
    };
}
 
export default withRouter(connect(mapStateToProps)(HomePage));