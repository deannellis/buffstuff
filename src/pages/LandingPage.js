import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { startLogin } from '../actions/auth';
import Button from '../components/Button';
import AppFooter from '../components/AppFooter';

const LoginPage = ({ startLogin }) => {
	return (
		<div className="landing-page">
			<div className="landing-page__hero">
				<div>
					<h1>flext takes the guess work out of weight lifting</h1>
					<Button clickHandler={startLogin()}>
						<img
							src="./images/64px-Google__G__Logo.webp"
							alt="Google mark"
							height="32px"
						/>
						Log in with Google
					</Button>
				</div>
				<div className="landing-page__hero-image">
					<img
						src="./images/landingIllustration.png"
						alt="illustration of weightlifter"
					/>
				</div>
			</div>
			<div className="landing-page__section-2">
				<div>
					<img
						src="./images/workoutComponents.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>
						flext knows what your next workout looks like so you don&apos;t have
						to
					</h1>
				</div>
			</div>
			<div className="landing-page__section-3">
				<div>
					<h1>See your gains increase as you progress</h1>
				</div>
				<div>
					<img
						src="./images/weightGraph.png"
						alt="Components from within the application"
					/>
				</div>
			</div>
			<div className="landing-page__section-4">
				<div>
					<img
						src="./images/macroComponent.png"
						alt="Components from within the application"
					/>
				</div>
				<div>
					<h1>Track your macros to optimize your gains</h1>
				</div>
			</div>
			<AppFooter />
		</div>
	);
};

const mapDispatchToProps = (dispatch) => ({
	startLogin: () => dispatch(startLogin),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
