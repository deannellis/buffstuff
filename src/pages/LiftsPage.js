import React, { Component } from 'react';
import { withRouter } from "react-router-dom";
import { connect } from 'react-redux';
import SideNav from '../components/SideNav';
import Tabs from '../components/Tabs';
import LineGraph from '../components/LineGraph';

class LiftsPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            activeTab: 0
        }
    }

    getData = () => {
        const { workouts } = this.props;
        let filteredWorkouts = [];
        workouts.map(workout => {
            let workoutWithDate = {}
            if(workout.hasOwnProperty('squat')) {
                workoutWithDate = {
                    ...workout['squat'],
                    date: workout.created
                }
                filteredWorkouts.push(workoutWithDate)
            }
        })
        // console.log('fw', filteredWorkouts);
        return filteredWorkouts
    }

    render() { 
        return (
            <div className="page--with-side-nav">
                <SideNav path={this.props.match.path} />
                <div className="lifts-page">
                    <Tabs 
                        activeIndex={this.state.activeTab} 
                        handleSelect={i => this.setState({ activeTab: i })}
                        labels={['Bench Press', 'Deadlift', 'Overhead Press', 'Row', 'Squat']}
                    >
                        <LineGraph data={this.getData()}/>

                        {/* {this.state.activeTab === 0 && 
                            <p>1</p>
                        }
                        {this.state.activeTab === 1 && 
                            <p>2</p>
                        }
                        {this.state.activeTab === 2 && 
                            <p>3</p>
                        }
                        {this.state.activeTab === 3 && 
                            <p>4</p>
                        }
                        {this.state.activeTab === 4 && 
                            <p>5</p>
                        } */}
                    </Tabs>
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        workouts: state.workouts,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftsPage));