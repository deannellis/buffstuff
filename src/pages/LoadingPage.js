import React from 'react';

const LoadingPage = () => {
	return (
		<div className="loading-page">
			<img alt="loading spinner" src="./images/Spinner-1s-200px.gif" />
		</div>
	);
};

export default LoadingPage;
