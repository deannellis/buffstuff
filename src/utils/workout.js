import moment from 'moment';

export const getWorkouts = ({ a = 0, b = 0 }) => {
    let workouts = []
    if(a == 0) {
        workouts.push('overhead', 'chinup')
    } else {
        workouts.push('bench', 'row')
    }
    if(b == 0 || b == 2) {
        workouts.push('squat');
    } else {
        workouts.push('deadlift');
    }
    return workouts;
}

export const getDisplayName = name => {
    switch(name) {
        case 'bench':
            return 'Bench Press';
        case 'row':
            return 'Row';
        case 'squat':
            return 'Squat';
        case 'deadlift':
            return 'Deadlift';
        case 'overhead':
            return 'Overhead Press';
        case 'chinup':
            return 'Chin-Ups';
        default:
            return name;
    }
};

export const getWarmupWeights = workWeight => {
    const warmup25 = workWeight * 0.25 > 44 ? workWeight * 0.25 : 44;
    const warmup50 = workWeight * 0.50 > 44 ? workWeight * 0.50 : 44;
    const warmup75 = workWeight * 0.75 > 44 ? workWeight * 0.75 : 44;
    return [ warmup25, warmup50, warmup75 ];
}

export const getWeightDistribution = weight => {
    if(weight === 44) {
        return [ 0, 44, 0]
    }
    const oneSide = (weight - 44) / 2;
    return [ oneSide, 44, oneSide ];
}

export const getMonthWorkouts = workouts => {
    let dates = {};
    workouts.forEach(workout => {
        let month = moment(workout.created).format("MMMM");
        let day = parseInt(moment(workout.created).format("D"));
        if(dates.hasOwnProperty(month)) {
            dates[month].push(day);
        } else {
            dates[month] = [ day ];
        }
    });
    // console.log(dates);
    return dates;
}

export const getWorkoutDays = (start, daysInMonth) => {
    let workoutDays = [];
    let checkEven = false;
    let weekday = start + 1;
    for(let d = 1; d <= daysInMonth; d++) {
        if(checkEven) {
            if(d%2 == 0) workoutDays.push(d)
        } else {
            if(d%2 !== 0) workoutDays.push(d)
        }
        if(weekday == 7){
            weekday = 0;
            checkEven = !checkEven;
        };
        weekday ++;
    }
    return workoutDays;
};
