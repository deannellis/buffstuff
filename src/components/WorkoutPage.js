import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from "react-router-dom";
import { getWorkouts, getDisplayName } from '../utils/workout';
import Button from '../components/Button';

class WorkoutPage extends Component {
    constructor(props) {
        super(props);
        console.log('Here ', props.inProgressWorkout[0]);
        this.state = {
             complete: false
        }
    }

    onStartLift = lift => {
        console.log(lift);
        this.props.history.push(`/workout/${lift}`);
    }

    componentDidMount() {
        let complete = true;
        const { inProgressWorkout } = this.props;
        for(let lift in inProgressWorkout) {
            if(inProgressWorkout[lift] === null) complete = false;
        }
        if(complete) {this.setState({complete});}
    }

    render() { 
        const { negatives, ups, weight } = this.props.masterWeights.chinup;
        return (
            <div className="workout-page">
                <h1>{`Workout #${this.props.workouts.length +1}`}</h1>
                {getWorkouts(this.props.liftVariant).map(lift => {
                    return (
                        <div key={lift} className="workout-page__lift">
                            <h2>
                                {getDisplayName(lift)}
                            </h2>
                            <p className="">{
                                lift !== 'chinup' ? 
                                `3 sets of 5 reps @${this.props.masterWeights[lift]}lbs` 
                                : 
                                `
                                    ${negatives === 0 ? '' : negatives + ' negatives'}
                                    ${negatives !== 0 && ups !== 0 ? ', ' : '' }
                                    ${ups === 0 ? '' : ups + ' chin-ups '}
                                    ${weight === 0 ? '' : weight + 'lbs'}

                                `    
                            }</p>
                            {this.props.inProgressWorkout[lift] !== null ? (
                                <p>
                                    Completed:  
                                    {this.props.inProgressWorkout[lift] === 0 && lift !== 'chinup' ? ' Last set less than 5 reps' : ''} 
                                    {this.props.inProgressWorkout[lift] === 1 && lift !== 'chinup' ? ' Last set greater than 5 reps' : ''} 
                                    {this.props.inProgressWorkout[lift] === 2 && lift !== 'chinup' ? ' Last set greater than 10 reps' : ''} 
                                    {this.props.inProgressWorkout[lift] === 0 && lift === 'chinup' ? ' Better luck next time' : ''} 
                                    {this.props.inProgressWorkout[lift] === 1 && lift === 'chinup' ? ' All reps completed!' : ''} 
                                </p>
                            ) : (
                                <Button variant="primary" clickHandler={() => {this.onStartLift(lift)}}>begin lift</Button>
                            )}
                        </div>
                    );
                })}
                {}
                {this.state.complete && 
                    <div className="workout-page__lift">
                        <h2>Hell Yeah!</h2>
                        <Button variant="primary">Finish Workout</Button>
                    </div>
                }
            </div>
        );
    }
}
 

const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
        liftVariant: state.liftVariant,
        workouts: state.workouts,
        inProgressWorkout: state.inProgressWorkout
    };
}
 
export default withRouter(connect(mapStateToProps)(WorkoutPage));