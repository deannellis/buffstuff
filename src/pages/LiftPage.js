import React, { Component } from 'react';
import { withRouter } from "react-router";
import { connect } from 'react-redux';
import { updateWorkout } from '../actions/inProgressWorkout';
import { getDisplayName, getWarmupWeights, getWeightDistribution } from '../utils/workout';
import Button from '../components/Button';

class LiftPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

    completeLift = (result) => {
        const liftResult = {updates: {[this.props.match.params.id]: result}}
        console.log(liftResult);
        this.props.dispatch(updateWorkout(liftResult));
        this.props.history.push('/workout');
    }

    render() { 
        let { id } = this.props.match.params;
        const workWeight = this.props.masterWeights[id];
        const workReps = ['5 reps', '5 reps', 'AMRAP*'];
        const { negatives, ups, weight } = this.props.masterWeights.chinup;
        return (
            <div className="lift-page">
                <div className="lift-page__card">
                    <h2>
                        {getDisplayName(id)}
                    </h2>
                    {id === 'chinup' ? (
                        <div>
                            <h3>Complete</h3>
                            <p className="lift-page__chinup-title">3 sets of</p>
                            <div className="lift-page__chinups">
                                {negatives === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{negatives}</p>
                                        <span>negatives</span>
                                    </div>
                                )}
                                {ups === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{ups}</p>
                                        <span>chin-ups</span>
                                    </div>
                                )}
                                {weight === 0 ? '' : (
                                    <div className="lift-page__chinup">
                                        <p>{weight}</p>
                                        <span>lbs</span>
                                    </div>
                                )}
                            </div>
                            <h3>Enter your results</h3>
                            <p>Were you able to complete all sets and reps?</p>
                            <Button variant="primary" clickHandler={() => {this.completeLift(1)}}>Yes</Button>
                            <Button clickHandler={() => {this.completeLift(0)}}>No</Button>
                        </div>
                    ) : (
                        <>
                            <div className="lift-page__tables">
                                <div>
                                    <h3>Warm-Up</h3>
                                    {getWarmupWeights(workWeight).map((weight, i) => (
                                        <div key={i} className="lift-page__sets">
                                            <div className="lift-page__set">
                                                <span>Set {i + 1}</span>
                                                <p>5 reps @{weight}lbs</p>
                                            </div>
                                            <div className="lift-page__distributed">
                                                {getWeightDistribution(weight).map((distWeight, i) => (
                                                    <div key={i} className="lift-page__distributed-item">
                                                        <span>
                                                            {i === 0 && 'LEFT'}
                                                            {i === 1 && 'BAR'}
                                                            {i === 2 && 'RIGHT'}
                                                        </span>
                                                        <p>{distWeight}</p>
                                                    </div>
                                                ))}
                                            </div>
                                        </div>
                                    ))}
                                </div>
                                <div>
                                    <h3>Work</h3>
                                    {workReps.map((rep, i) => (
                                        <div key={i} className="lift-page__sets">
                                                <div className="lift-page__set">
                                                    <span>Set {i + 1}</span>
                                                    <p>{rep} @{workWeight}lbs</p>
                                                </div>
                                                <div className="lift-page__distributed">
                                                    {getWeightDistribution(workWeight).map((distWeight, i) => (
                                                        <div key={i} className="lift-page__distributed-item">
                                                            <span>
                                                                {i === 0 && 'LEFT'}
                                                                {i === 1 && 'BAR'}
                                                                {i === 2 && 'RIGHT'}
                                                            </span>
                                                            <p>{distWeight}</p>
                                                        </div>
                                                    ))}
                                                </div>
                                        </div>
                                    ))}
                                </div>
                            </div>
                            <p>*as many reps as possible </p>
                            <h3 className="lift-page__results-title">Enter your results</h3>
                            <p>On your last set how many reps did you complete?</p>
                            <Button clickHandler={() => {this.completeLift(0)}}>Less than 5</Button>
                            <Button clickHandler={() => {this.completeLift(1)}}>Greater than 5</Button>
                            <Button clickHandler={() => {this.completeLift(2)}}>Greater than 10</Button>
                        </>
                    )}
                    
                </div>
            </div>
        );
    }
}
 
const mapStateToProps = state => {
    return {
        masterWeights: state.masterWeights,
    };
}
 
export default withRouter(connect(mapStateToProps)(LiftPage));