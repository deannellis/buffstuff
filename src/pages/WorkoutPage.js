import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import moment from 'moment';
import { Trail } from 'react-spring/renderprops';

import { getWorkouts, getDisplayName, getEmoji } from '../utils/workout';
import Button from '../components/Button';
import { startAddWorkout } from '../actions/workouts';
import { resetWorkout, startWorkout } from '../actions/inProgressWorkout';
import { startUpdateMasterWeights } from '../actions/masterWeights';
import { startUpdateLiftVariant } from '../actions/liftVariant';
import { LeftArrowIcon } from '../utils/icons';
import MenuContext from '../context/menu-context';

export class WorkoutPage extends Component {
	constructor(props) {
		super(props);
		this.state = {
			complete: false,
			created: moment(),
		};
	}

	componentDidMount() {
		let complete = true;
		const { inProgressWorkout, onStartWorkout } = this.props;
		const { setPageMenu } = this.context;
		const workoutKeys = Object.keys(inProgressWorkout);
		if (workoutKeys.length === 0) {
			complete = false;
			onStartWorkout();
		}
		workoutKeys.forEach((lift) => {
			if (inProgressWorkout[lift] === null) complete = false;
		});
		if (complete) {
			this.setState({ complete });
		}
		setPageMenu(false);
	}

	componentWillUnmount() {
		const { setPageMenu } = this.context;
		setPageMenu(true);
	}

	onStartLift = (lift) => {
		const { history } = this.props;
		history.push(`/workout/${lift}`);
	};

	completeWorkout = () => {
		const {
			inProgressWorkout,
			masterWeights,
			history,
			onStartAddWorkout,
			onUpdateMasterWeights,
			onUpdateLiftVariant,
			onResetWorkout,
		} = this.props;
		const { created } = this.state;
		const workoutPayload = {
			workout: inProgressWorkout,
			currentWeight: masterWeights,
			created: created.valueOf(),
		};
		onStartAddWorkout(workoutPayload);
		onUpdateMasterWeights(inProgressWorkout);
		onUpdateLiftVariant();
		onResetWorkout();
		history.push(`/home`);
	};

	render() {
		const {
			masterWeights,
			workouts,
			liftVariant,
			inProgressWorkout,
			history,
		} = this.props;
		const { complete } = this.state;
		const { negatives, ups, weight } = masterWeights.chinup;
		const emojis = [getEmoji(), getEmoji(), getEmoji()];
		return (
			<div className="workout-page__wrapper">
				<div className="workout-page">
					<h1>{`Workout #${workouts.length + 1}`}</h1>
					{complete && (
						<div className="workout-page__lift card">
							<h2>Hell Yeah!</h2>
							<Button variant="primary" clickHandler={this.completeWorkout}>
								Finish Workout
							</Button>
						</div>
					)}
					<Trail
						items={getWorkouts(liftVariant)}
						keys={(lift) => lift}
						from={{ opacity: 0, transform: 'translate3d(0,-40px,0)' }}
						to={{ opacity: 1, transform: 'translate3d(0,0px,0)' }}
					>
						{(lift) => (props) => (
							<div className="workout-page__lift card" style={props}>
								<h2>{getDisplayName(lift)}</h2>
								<p>
									{lift !== 'chinup'
										? `3 sets of 5 reps @${masterWeights[lift]}lbs`
										: `
                                        ${
																					negatives === 0
																						? ''
																						: `${negatives} negatives`
																				}
                                        ${
																					negatives !== 0 && ups !== 0
																						? ', '
																						: ''
																				}
                                        ${ups === 0 ? '' : `${ups} chin-ups`}
                                        ${weight === 0 ? '' : `${weight} lbs`}

                                    `}
								</p>
								{inProgressWorkout[lift] !== null ? (
									<p>
										Completed:
										{inProgressWorkout[lift] === 0 && lift !== 'chinup'
											? ' Last set less than 5 reps'
											: ''}
										{inProgressWorkout[lift] === 1 && lift !== 'chinup'
											? ' Last set greater than 5 reps '
											: ''}
										{inProgressWorkout[lift] === 1 && lift !== 'chinup' && (
											<span>{emojis[0]}</span>
										)}
										{inProgressWorkout[lift] === 2 && lift !== 'chinup'
											? ' Last set greater than 10 reps '
											: ''}
										{inProgressWorkout[lift] === 2 && lift !== 'chinup' && (
											<span>{emojis[1]}</span>
										)}
										{inProgressWorkout[lift] === 0 && lift === 'chinup'
											? ' Better luck next time'
											: ''}
										{inProgressWorkout[lift] === 1 && lift === 'chinup'
											? ' All reps completed! '
											: ''}
										{inProgressWorkout[lift] === 1 && lift === 'chinup' && (
											<span>{emojis[2]}</span>
										)}
									</p>
								) : (
									<Button
										variant="primary"
										clickHandler={() => {
											this.onStartLift(lift);
										}}
									>
										begin lift
									</Button>
								)}
							</div>
						)}
					</Trail>
					<button
						className="lift-page__action-button action-button"
						type="button"
						onClick={() => {
							history.goBack();
						}}
					>
						<LeftArrowIcon fill="rgba(26, 33, 46, .84)" size={24} />
						<span>Back</span>
					</button>
				</div>
			</div>
		);
	}
}
WorkoutPage.contextType = MenuContext;
WorkoutPage.propTypes = {
	masterWeights: PropTypes.shape({
		bench: PropTypes.number,
		row: PropTypes.number,
		squat: PropTypes.number,
		deadlift: PropTypes.number,
		overhead: PropTypes.number,
		chinup: PropTypes.object,
	}),
	liftVariant: PropTypes.shape({
		a: PropTypes.number,
		b: PropTypes.number,
	}),
	workouts: PropTypes.arrayOf(PropTypes.object),
	inProgressWorkout: PropTypes.objectOf(PropTypes.number),
	history: PropTypes.shape({
		push: PropTypes.func,
		goBack: PropTypes.func,
	}).isRequired,
	onStartWorkout: PropTypes.func.isRequired,
	onStartAddWorkout: PropTypes.func.isRequired,
	onUpdateMasterWeights: PropTypes.func.isRequired,
	onUpdateLiftVariant: PropTypes.func.isRequired,
	onResetWorkout: PropTypes.func.isRequired,
};
WorkoutPage.defaultProps = {
	masterWeights: {
		bench: 0,
		row: 0,
		squat: 0,
		deadlift: 0,
		overhead: 0,
		chinup: {
			negatives: 0,
			ups: 0,
			weight: 0,
		},
	},
	liftVariant: { a: 0, b: 0 },
	workouts: [],
	inProgressWorkout: {
		bench: null,
		row: null,
		squat: null,
	},
};

const mapStateToProps = (state) => {
	return {
		masterWeights: state.masterWeights,
		liftVariant: state.liftVariant,
		workouts: state.workouts,
		inProgressWorkout: state.inProgressWorkout,
	};
};
const mapDispatchToProps = (dispatch) => ({
	onStartWorkout: () => {
		dispatch(startWorkout());
	},
	onStartAddWorkout: (workout) => {
		dispatch(startAddWorkout(workout));
	},
	onUpdateMasterWeights: (inProgressWorkout) => {
		dispatch(startUpdateMasterWeights(inProgressWorkout));
	},
	onUpdateLiftVariant: () => {
		dispatch(startUpdateLiftVariant());
	},
	onResetWorkout: () => {
		dispatch(resetWorkout());
	},
});

export default withRouter(
	connect(mapStateToProps, mapDispatchToProps)(WorkoutPage)
);
