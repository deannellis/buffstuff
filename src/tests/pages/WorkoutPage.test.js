import React from "react";
import PropTypes from "prop-types";
import { shallow } from "enzyme";

import { WorkoutPage } from "../../pages/WorkoutPage";
import { workouts, weights } from "../fixtures/workout";

WorkoutPage.contextTypes = {
	setPageMenu: PropTypes.func
};
const context = { setPageMenu: () => {} };
test("should render WorkoutPage without props", () => {
	const wrapper = shallow(<WorkoutPage />, { context });
	expect(wrapper).toMatchSnapshot();
});

test("should render WorkoutPage with props", () => {
	const inProgressWorkout = { bench: 2, row: 2, squat: 2 };
	const wrapper = shallow(
		<WorkoutPage
			workouts={workouts}
			masterWeights={weights}
			liftVariant={{ a: 1, b: 1 }}
			inProgressWorkout={inProgressWorkout}
		/>,
		{ context }
	);
	expect(wrapper).toMatchSnapshot();
});
