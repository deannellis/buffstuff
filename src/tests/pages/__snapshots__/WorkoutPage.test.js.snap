// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WorkoutPage with props 1`] = `
<div
  className="workout-page__wrapper"
>
  <div
    className="workout-page"
  >
    <h1>
      Workout #4
    </h1>
    <div
      className="workout-page__lift card"
    >
      <h2>
        Hell Yeah!
      </h2>
      <Button
        clickHandler={[Function]}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Finish Workout
      </Button>
    </div>
    <Component
      from={
        Object {
          "opacity": 0,
          "transform": "translate3d(0,-40px,0)",
        }
      }
      items={
        Array [
          "bench",
          "row",
          "deadlift",
        ]
      }
      keys={[Function]}
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0,0px,0)",
        }
      }
    >
      <Component />
    </Component>
    <button
      className="lift-page__action-button action-button"
      onClick={[Function]}
    >
      <LeftArrowIcon
        fill="rgba(26, 33, 46, .84)"
        size={24}
      />
      <span>
        Back
      </span>
    </button>
  </div>
</div>
`;

exports[`should render WorkoutPage without props 1`] = `
<div
  className="workout-page__wrapper"
>
  <div
    className="workout-page"
  >
    <h1>
      Workout #1
    </h1>
    <Component
      from={
        Object {
          "opacity": 0,
          "transform": "translate3d(0,-40px,0)",
        }
      }
      items={
        Array [
          "overhead",
          "chinup",
          "squat",
        ]
      }
      keys={[Function]}
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0,0px,0)",
        }
      }
    >
      <Component />
    </Component>
    <button
      className="lift-page__action-button action-button"
      onClick={[Function]}
    >
      <LeftArrowIcon
        fill="rgba(26, 33, 46, .84)"
        size={24}
      />
      <span>
        Back
      </span>
    </button>
  </div>
</div>
`;
